core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -50,11 +55,12 @@ class ImprovementPickerScreen(tileInfo: TileInfo, onAccept: ()->Unit) : PickerSc
 
             group.add(image).size(30f).pad(10f)
 
-            var labelText = improvement.name.tr() + " - " + improvement.getTurnsToBuild(currentPlayerCiv) + " {turns}"
+            var labelText = improvement.name.tr()
+            improvement.getTurnsToBuild(currentPlayerCiv).let { if (it > 0) labelText += " - $it {turns}" }
```

Comment:
Please separate turnsToBuild into a val, and then use a regular if() to check if to add the string

Reply:
Yes, you told us not do overdo Kotlin cleverness. I try - clever where it definitely improves readability, old-fashioned where not. Bytecode will be the same anyway.